StackExchange.Redis/StackExchange/Redis/ResultProcessor.cs

```diff
@@ -1301,6 +1325,461 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
+                var consumerCount = (int)arr[3].AsRedisValue();
+                var pendingCount = (int)arr[5].AsRedisValue();
+
+                return new StreamGroupInfo(groupName, consumerCount, pendingCount);
+            }
+        }
+
+        internal abstract class InterleavedStreamInfoProcessorBase<T> : ResultProcessor<T[]>
+        {
+            protected abstract T ParseItem(RawResult result);
+
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, RawResult result)
+            {
+                if (result.Type != ResultType.MultiBulk)
+                {
+                    return false;
+                }
+
+                var arr = result.GetItems();
+                var parsedItems = new T[arr.Length];
+
+                for (var i = 0; i < arr.Length; i++)
+                {
+                    parsedItems[i] = ParseItem(arr[i]);
+                }
+
+                SetResult(message, parsedItems);
+                return true;
+            }
+        }
+
+        internal sealed class StreamInfoProcessor : StreamProcessorBase<StreamInfo>
+        {
+            // Parse the following format:
+            // > XINFO mystream
+            // 1) length
+            // 2) (integer) 13
+            // 3) radix-tree-keys
+            // 4) (integer) 1
+            // 5) radix-tree-nodes
+            // 6) (integer) 2
+            // 7) groups
+            // 8) (integer) 2
+            // 9) first-entry
+            // 10) 1) 1524494395530-0
+            //     2) 1) "a"
+            //        2) "1"
+            //        3) "b"
+            //        4) "2"
+            // 11) last-entry
+            // 12) 1) 1526569544280-0
+            //     2) 1) "message"
+            //        2) "banana"
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, RawResult result)
+            {
+                if (result.Type != ResultType.MultiBulk)
+                {
+                    return false;
+                }
+
+                var arr = result.GetItems();
+
+                if (arr.Length != 12)
+                {
+                    return false;
+                }
+
+                // The first four items are interleaved name/value pairs.
+                // Skip the field name and just get the value.
+
+                var length = (int)arr[1].AsRedisValue();
+                var treeKeyCount = (int)arr[3].AsRedisValue();
+                var treeNodeCount = (int)arr[5].AsRedisValue();
+                var groupCount = (int)arr[7].AsRedisValue();
+
+                // Note: Even if there is only 1 message in the stream, this command returns
+                //       the single entry as the first-entry and last-entry in the response.
+
+                var entries = ParseRedisStreamEntries(new RawResult(new RawResult[]
+                {
+                    arr[9],
+                    arr[11]
+                }));
```

Comment:
This one is a bit round-about to get the parsed result - we should probably break up the parsing so it's accessible/this is more direct without the intermediate creations.

Reply:
Something like this?

![image](https://user-images.githubusercontent.com/1170315/41945113-6a899418-7979-11e8-9dd3-26c112027a98.png)

Reply:
Something like this? 

![image](https://user-images.githubusercontent.com/1170315/41945194-cd89f0ee-7979-11e8-98a9-527154cc5072.png)

Reply:
This looks cleaner than the previous version. Another option would be to specify the constructor parameters by name instead of with the comment. I like to have an explicit identifier (named param or comment) of what it is I'm passing if it's not readily apparent from the value being passed (just a personal preference).